tests/test_numeric/test_eulerian_grid_ops/test_stencil_ops_3d/test_laplacian_filter_mpi_3d.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+import pytest
+from sopht.utils.precision import get_real_t, get_test_tol
+from sopht.numeric.eulerian_grid_ops.stencil_ops_3d import (
+    gen_laplacian_filter_kernel_3d,
+)
+from sopht_mpi.utils import (
+    MPIConstruct3D,
+    MPIGhostCommunicator3D,
+    MPIFieldCommunicator3D,
+)
+from sopht_mpi.numeric.eulerian_grid_ops.stencil_ops_3d import (
+    gen_laplacian_filter_mpi_kernel_3d,
+)
+
+
+@pytest.mark.mpi(group="MPI_stencil_ops_3d", min_size=4)
+@pytest.mark.parametrize("ghost_size", [1, 2])
+@pytest.mark.parametrize("precision", ["single", "double"])
+@pytest.mark.parametrize(
+    "rank_distribution",
+    [(0, 1, 1), (1, 0, 1), (1, 1, 0), (1, 0, 0), (0, 1, 0), (0, 0, 1)],
+)
+@pytest.mark.parametrize("aspect_ratio", [(1, 1, 1), (1, 1.5, 2)])
+@pytest.mark.parametrize("field_type", ["scalar", "vector"])
+@pytest.mark.parametrize("filter_type", ["multiplicative", "convolution"])
+def test_mpi_laplacian_filter_3d(
+    ghost_size, precision, rank_distribution, aspect_ratio, field_type, filter_type
+):
+    n_values = 8
+    grid_size_z, grid_size_y, grid_size_x = (n_values * np.array(aspect_ratio)).astype(
+        int
+    )
+    real_t = get_real_t(precision)
+    # Generate the MPI topology minimal object
+    mpi_construct = MPIConstruct3D(
+        grid_size_z=grid_size_z,
+        grid_size_y=grid_size_y,
+        grid_size_x=grid_size_x,
+        real_t=real_t,
+        rank_distribution=rank_distribution,
+    )
+
+    # extra width needed for kernel computation
+    mpi_ghost_exchange_communicator = MPIGhostCommunicator3D(
+        ghost_size=ghost_size, mpi_construct=mpi_construct
+    )
+    mpi_field_communicator = MPIFieldCommunicator3D(
+        ghost_size=ghost_size, mpi_construct=mpi_construct
+    )
+    gather_local_field = (
+        mpi_field_communicator.gather_local_scalar_field
+        if field_type == "scalar"
+        else mpi_field_communicator.gather_local_vector_field
+    )
+    scatter_global_field = (
+        mpi_field_communicator.scatter_global_scalar_field
+        if field_type == "scalar"
+        else mpi_field_communicator.scatter_global_vector_field
+    )
```

Comment:
`match-case` would be apt here as well.

Reply:
Similarly all case-based switches in the following code.

Reply:
Fixed in [0e17b87](https://github.com/fankiat/sopht-mpi/pull/148/commits/0e17b87002a91fb7978b3e815bdb78a8dfa577e4).